source/user/df.rst

```diff
@@ -5,3 +5,143 @@ Density fitting
 
 *Modules*: :mod:`df`, :mod:`pbc.df`
 
+.. module:: df
+   :synopsis: Density fitting and RI approximation
+.. sectionauthor:: Qiming Sun <osirpt.sun@gmail.com>.
+
+Introduction
+============
+
+Density fitting (DF), sometimes also called the resolution of identity (RI) approximation, is a method to approximate the four-index electron repulsion integrals (ERIs) by two- and three-index tensors. :cite:`Dunlap00PCCP` It achieves so by expanding the atomic orbital (AO) pair density in a second, auxiliary Gaussian basis set. PySCF has extensive supports for using DF in both molecular and crystalline calculations. This section of the user document covers the basic usage of DF in molecular applications. See :ref:`theory_pbc_df` for the use of DF in periodic calculations.
+
+
+Using DF in calculations
+========================
+
+DF is not used by default when initializing a :class:`SCF` or :class:`MCSCF` object, but can be invoked via the :func:`density_fit` method as follows::
+
+    from pyscf import gto, scf, mcscf
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mf = scf.RHF(mol).density_fit().run()
+    mc = mcscf.CASSCF(mf, 8, 10).density_fit().run()
+
+Alternatively, one can initialize a :class:`DF` object and overwrite the :attr:`with_df` attribute of the :class:`SCF` object::
+
+    from pyscf import gto, scf, df
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mydf = df.DF(mol).build()
+    mf = scf.RHF(mol).density_fit()
+    mf.with_df = mydf
+    mf.kernel()
+
+Setting :attr:`with_df` to ``None`` turns DF off.
+
+Once the SCF calculations using DF are done, all post-SCF methods will use DF automatically. The example below performs a TDDFT/TDA calculation using DF::
+
+    from pyscf import gto, dft, tddft
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mf = dft.RKS(mol, xc="b3lyp").density_fit().run()
+    td = tddft.TDA(mf).run()
+    print(td.e)
```

Comment:
What's the auxiliary basis that ends up used here?

Reply:
Good question! I checked the source code of the TDDFT module and didn't find any trace of DF being used explicitly. For example, when ERI is needed as in [this line](https://github.com/pyscf/pyscf/blob/master/pyscf/tdscf/rhf.py#L137), it directly calls the `ao2mo`.

Reply:
I just DM-ed qiming on this and will make revisions accordingly.